File: python/204.py
xrange = range


class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        isprime = [True] * n
        # isprime[0] = isprime[1] = False
        for i in xrange(2, int(n ** 0.5) + 1):
            if isprime[i]:
                for j in xrange(i*i, n, i):
                    isprime[j] = False
        return sum(isprime[2:])

File: cpp/204.cpp
#include <vector>
#include <cmath>
using namespace std;

class Solution {
public:
  int countPrimes(int n) {
    vector<bool> isprime(n, true);
    int upbound = ceil(sqrt(n));
    for(int i=2; i<=upbound; i++){
      if(isprime[i])
        for(int j=i*i; j < n; j+=i)
          isprime[j] = false;
    }
    int ans = 0;
    for(int i=2; i<n; i++) ans += isprime[i];
    return ans;
  }
};
